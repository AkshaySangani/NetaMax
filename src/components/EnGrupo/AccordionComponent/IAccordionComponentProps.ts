export interface IAccordionItemComponentProps {
  /**
   * Title for accordion.
   * @type {string}
   */
  title: string;

  /**
   * Products for accordion.
   * @type {Array<{name: string}}}
   */
  products: Array<{ name: string }>;

  /**
   * Page for accordion pagination.
   * @type {number}
   */
  page: number;

  /**
   * TotalPage for accordion pagination.
   * @type {number}
   */
  totalPage: number;
}

export interface IAccordionComponentProps {
  /**
   * Items for accordion.
   * @type {Array<IAccordionItemComponentProps>}
   */
  items: Array<IAccordionItemComponentProps>;

  /**
   * OnChangePagination for accordion pagination.
   * @type {(categoryIndex: number, page: number) => void}
   */
  onChangePagination: (categoryIndex: number, page: number) => void;
}
