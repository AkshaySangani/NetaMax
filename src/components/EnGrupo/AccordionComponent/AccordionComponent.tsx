import React, { ReactElement } from "react";

import ArrowBottomBorder from "styled/icons/ArrowBottomBorder";
import ArrowTopBorder from "styled/icons/ArrowTopBorder";

import {
  Accordion,
  AccordionButton,
  AccordionItem,
  AccordionPanel,
  Box,
  Text,
} from "@chakra-ui/react";

import PaginationComponent from "../PaginationComponent/PaginationComponent";
import { IAccordionComponentProps } from "./IAccordionComponentProps";

/**
 * The AccordionComponent component.
 * @param {IAccordionComponentProps} props the component props.
 * @returns {ReactElement} the component element.
 */
const AccordionComponent = (props: IAccordionComponentProps): ReactElement => {
  const { items, onChangePagination } = props;
  const [expanded, setExpanded] = React.useState<number>();

  /**
   * The handleChangeAccordion.
   * @param {Array<number>} index the component props.
   * @returns {void} the component element.
   */
  const handleChangeAccordion = (index: number): void => {
    setExpanded(index);
  };

  return (
    <Accordion allowToggle onChange={handleChangeAccordion}>
      {items.map((item, indexCategory) => {
        return (
          <AccordionItem key={indexCategory} borderTop={0} borderBottom="1px solid #E4E4E4">
            <AccordionButton padding="1rem 0">
              <Box
                flex="1"
                textAlign="left"
                fontWeight="bold"
                fontSize={24}
                lineHeight="30.72px"
                color="#242424"
              >
                {item.title}
              </Box>
              <Text fontSize={16} fontWeight={600} marginRight="1rem" textColor="#F04D4B">
                {expanded === indexCategory ? "Ocultar" : "Ver"}
              </Text>
              {expanded === indexCategory ? <ArrowTopBorder /> : <ArrowBottomBorder />}
            </AccordionButton>
            <AccordionPanel padding={0}>
              {item.products.map((item, index) => {
                return <Box key={index}>{item.name}</Box>;
              })}
              {item.totalPage > 1 && (
                <PaginationComponent
                  page={item.page}
                  totalPage={item.totalPage}
                  onChangePage={(page) => onChangePagination(indexCategory, page)}
                />
              )}
            </AccordionPanel>
          </AccordionItem>
        );
      })}
    </Accordion>
  );
};

export default AccordionComponent;
