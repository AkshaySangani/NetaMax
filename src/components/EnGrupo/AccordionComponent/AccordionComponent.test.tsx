import { render, fireEvent } from "@testing-library/react";
import AccordionComponent from "./AccordionComponent";
import { IAccordionItemComponentProps } from "./IAccordionComponentProps";

jest.mock("next/router", () => require("next-router-mock"));

describe("Render AccordionComponent component", () => {
  it("AccordionComponent should be render correctly", () => {
    const { container } = render(<AccordionComponent items={[]} onChangePagination={() => null} />);
    expect(container).toMatchSnapshot();
  });

  it("AccordionComponent should click in page, when expanded category", () => {
    const onClickMock = jest.fn();
    const items: IAccordionItemComponentProps[] = [
      {
        title: "Lo más vendido",
        products: [
          {
            name: "Leche Clasica Alpura",
          },
        ],
        page: 1,
        totalPage: 5,
      },
    ];
    const { getByText } = render(
      <AccordionComponent items={items} onChangePagination={onClickMock} />
    );
    fireEvent.click(getByText("Lo más vendido"));
    fireEvent.click(getByText("2"));
    expect(onClickMock).toBeCalledTimes(1);
  });
});
