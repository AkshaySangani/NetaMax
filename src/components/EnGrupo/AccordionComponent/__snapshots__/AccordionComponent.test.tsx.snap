// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render AccordionComponent component AccordionComponent should be render correctly 1`] = `
<div>
  <div
    class="chakra-accordion css-0"
  />
</div>
`;
